nntrainer/src/tensor.cpp

```diff
@@ -673,14 +685,37 @@ void Tensor::read(std::ifstream &file) {
 /**
  * @brief Calculate average value according to the axis.
  */
-Tensor Tensor::average(int axis) const {
+Tensor Tensor::average(unsigned int axis) const {
+  if (axis >= MAXDIM)
+    throw std::out_of_range(
+      "negative axis or axis more then MAXDIM is invalid");
+
   unsigned int axis_size = dim.getDim()[axis];
   if (axis_size == 1)
     return this->clone();
 
   return this->sum(axis, 1.0 / ((float)axis_size));
 }
 
+Tensor Tensor::average(const std::vector<unsigned int> &axes) const {
+  if (axes.size() > MAXDIM)
+    throw std::invalid_argument("too many axes given");
+
+  if (axes.empty())
+    return this->average();
+
+  unsigned int axes_size = 1;
+  for (const auto idx : axes) {
+    if (idx >= MAXDIM) {
```

Comment:
Create a new dimension where the given axes will be set to 1.
Then use `axes_size = dim.getDataLen() / newDim.getDataLen()`
This ensures that even though an axes is passed twice, it will produce correct answer.